src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -5311,6 +5313,365 @@ class StubGenerator: public StubCodeGenerator {
+                                                : Assembler::INVALID_ARRANGEMENT;
+    guarantee(load_arrangement != Assembler::INVALID_ARRANGEMENT, "invalid arrangement");
+
+    const int multiply_by_halves =
+        load_arrangement == Assembler::T4S || load_arrangement == Assembler::T4H   ? false
+        : load_arrangement == Assembler::T8B || load_arrangement == Assembler::T8H ? true
+                                                                                   : -1;
+    guarantee(multiply_by_halves != -1, "unknown multiplication algorithm");
+
+    const int small_loop_size = load_arrangement == Assembler::T4S   ? 20  // 5 insts
+                                : load_arrangement == Assembler::T4H ? 24  // 6 insts
+                                : load_arrangement == Assembler::T8H ? 36  // 9 insts
+                                : load_arrangement == Assembler::T8B ? 40  // 10 insts
+                                                                     : -1; // invalid
+    guarantee(small_loop_size != -1, "invalid small_loop_size");
+
+    // Put 0-3'th powers of 31 into a single SIMD register together. The register will be used in
+    // the SMALL and LARGE LOOPS' epilogues. The initialization is hoisted here and the register's
+    // value shouldn't change throughout both loops.
+    __ movw(rscratch1, intpow(31U, 3));
+    __ mov(vpow, Assembler::S, 0, rscratch1);
+    __ movw(rscratch1, intpow(31U, 2));
+    __ mov(vpow, Assembler::S, 1, rscratch1);
+    __ movw(rscratch1, intpow(31U, 1));
+    __ mov(vpow, Assembler::S, 2, rscratch1);
+    __ movw(rscratch1, intpow(31U, 0));
+    __ mov(vpow, Assembler::S, 3, rscratch1);
+
+    __ mov(vmul0, Assembler::T16B, 0);
+    __ mov(vmul0, Assembler::S, 3, result);
+
+    __ andr(rscratch2, cnt, (uf - 1) * vf);
+    __ cbz(rscratch2, LARGE_LOOP_PREHEADER);
+
+    __ movw(rscratch1, intpow(31U, multiply_by_halves ? vf / 2 : vf));
+    __ mov(vpowm, Assembler::S, 0, rscratch1);
+
+    if (small_loop_size % 32 > 32 - __ offset() % 32) {
+      __ align(32);
+    }
+
+    auto start = __ offset();
+    __ bind(SMALL_LOOP);
+
+    __ ld1(vdata0, load_arrangement, Address(__ post(ary, vf * type2aelembytes(eltype))));
+    __ mulv(vmul0, Assembler::T4S, vmul0, vpowm, 0);
+    __ subsw(rscratch2, rscratch2, vf);
+
+    if (load_arrangement == Assembler::T8B) {
+      // Extend 8B to 8H to be able to use vector multiply
+      // instructions
+      assert(load_arrangement == Assembler::T8B, "expected to extend 8B to 8H");
+      if (is_signed_subword_type(eltype)) {
+        __ sxtl(vdata0, Assembler::T8H, vdata0, load_arrangement);
+      } else {
+        __ uxtl(vdata0, Assembler::T8H, vdata0, load_arrangement);
+      }
+    }
+
+    if (load_arrangement == Assembler::T4S) {
+      __ addv(vmul0, load_arrangement, vmul0, vdata0);
+    } else if (load_arrangement == Assembler::T8B || load_arrangement == Assembler::T4H ||
+               load_arrangement == Assembler::T8H) {
```

Comment:
Use a switch here, and everywhere else that a switch applies.

Reply:
The only other piece of code similar to this one is the one at line [5591](https://github.com/openjdk/jdk/pull/18487/files#diff-9112056f732229b18fec48fb0b20a3fe824de49d0abd41fbdb4202cfe70ad114R5591). Any other which I'm missing?

Reply:
Fixed by https://github.com/openjdk/jdk/pull/18487/commits/a824a74263ce9309fca50a90b3d0f09e9b56a5c4 , please check.